test/format/format.sh

```diff
@@ -449,7 +456,7 @@ format()
 
 	# Disassociate the command from the shell script so we can exit and let the command
 	# continue to run.
-	# Run format in its own session so child processes are in their own process gorups
+	# Run format in its own session so child processes are in their own process groups
 	# and we can individually terminate (and clean up) running jobs and their children.
 	nohup setsid $cmd > $log 2>&1 &
 
```

Comment:
Are lines 463-471 (checking log file for signature 'setsid: failed to execute' and marking failure) made redundant with this solution?

Reply:
I think it's still needed.

I think there's a window where we don't handle things correctly, in-between calling `setsid` and seeing`running process` in the log file. I moved the code in `format` that writes that line to the log file to be almost the first thing `format` does, but if for some reason the exec just fails, we wouldn't handle it correctly. And, this line tries to get that correct by looking for an exec failure.

There's going to be a window no matter what we do, but I think we've tightened it down pretty hard.

Reply:
Thanks for the explanation.